src/main/java/net/neoforged/neoforge/registries/DeferredRegister.java

```diff
@@ -164,7 +178,9 @@ public static DeferredRegister.Blocks createBlocks(String modid) {
      * @see #create(ResourceKey, String)
      * @see #create(ResourceLocation, String)
      * @see #createItems(String)
+     * @deprecated Use {@link DeferredRegister#createDataComponents(ResourceKey, String)}
```

Comment:
Per discussion [in Discord](https://discord.com/channels/313125603924639766/1154167065519861831/1282456685008523399), please add in the `@deprecated` javadoc the version in which this method will be removed. Additionally, for ease of transition, the registry key for the item data components should be mentioned as well. (Please double-check my reference is right.)

```suggestion
     * @deprecated To be removed in 1.21.2; use {@link DeferredRegister#createDataComponents(ResourceKey, String)} with {@link Registries#DATA_COMPONENT_TYPE} instead.
```

(Same comment for the other javadoc.)